src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -204,9 +204,30 @@ write_cb(char *data, size_t size, size_t nmemb, void *connp)
         SOL_STR_SLICE_STR(data, data_size));
     SOL_INT_CHECK(r, < 0, 0);
 
+    if (connection->interface->write_cb) {
+        data_size = connection->interface->write_cb((void *)connection->data, connection, &connection->buffer);
+    }
+
+
     return data_size;
 }
 
+static size_t
+read_cb(char *buffer, size_t size, size_t nitems, void *connp)
+{
+    struct sol_http_client_connection *connection = connp;
+    size_t data_size, ret;
+    int r;
+
+    r = sol_util_size_mul(size, nitems, &data_size);
+    SOL_INT_CHECK(r, < 0, 0);
+
+    ret = connection->interface->read_cb((void *)connection->data,
+        connection, buffer, data_size);
```

Comment:
As you explained to me in person, this should not be const, sure. But we can still go with memory-not-owned/fixed-capacity.

as I said in person, I think the method names are confusing. We shouldn't mimic cURL, just try to match the action user expects to do (like in "write_cb" you expect user to receive some data, in "read_cb" you expect user to send some data)